### PR TITLE
LPS-87074 Avoid NPE when job is unscheduled on master node

### PR DESCRIPTION
**Pull request by istvansajtos**
[LPS-87074](https://issues.liferay.com/browse/LPS-87074)

Hi Eric,

The issue is that slave node throws NullPointerException if there is an unscheduled job on master. This is very similar to [LPS-77562](https://issues.liferay.com/browse/LPS-77562) and I used the same approach to fix it.

Please review my change and let me know if there is anything I can help you with.

Thank you,
Istvan

### CODE REVIEW

**Comment by istvansajtos** [comment]
ci:test:sf

**Comment by istvansajtos** [comment]
ci:test:relevant

**Comment by ericyanLr** [comment]
Hi @istvansajtos ,

Looking at the changes, is the **NullPointerException** being thrown, due to the **oldTrigger** being null?

If so, can we verify that the NullPointerException is being thrown, due to the removal of the trigger, which is intentionally being done by using the script script in the Steps to Reproduce?

If this is the case, can we further refine the code changes to better reflect this, as well as provide a more accurate log message?

Please let me know your thoughts.
Thanks!

**Comment by istvansajtos** [comment]
Hi Eric,

Thanks for sharing your ideas.

> Looking at the changes, is the **NullPointerException** being thrown, due to the **oldTrigger** being null?

Yes, exactly.

> If so, can we verify that the NullPointerException is being thrown, due to the removal of the trigger, which is intentionally being done by using the script script in the Steps to Reproduce?
> 
> If this is the case, can we further refine the code changes to better reflect this, as well as provide a more accurate log message?

Having separate null checks for *schedulerResponse* and *schedulerResponse.getTrigger()* and thus having distinctive logging for those cases is technically possible. What I am not sure is what the log messages should contain exactly. If you have anything specific in mind regarding how the proper messages should look like, please share with me.

I will update the PR with the requested changes soon.

Regards,
Istvan

**Comment by ericyanLr** [comment]
Hi @istvansajtos ,

It sounds like currently, we are fixing a bug that is encountered by an edge case. 

**Null Check**
Looking at the code and the exception described on the ticket, it looks like the **NullPointerException** is being caused by the **trigger object** being null.

But, I haven't actually investigated this issue, so I don't know for sure if we would ever expect the **schedulerResponse** object to be null. From my understanding, the current logic does not expect that object to be null.

If you think that there is a chance that the **schedulerResponse** would ever be null, after your investigation into this issue, then I think it would be reasonable to add that null check. I just want to make sure we're not adding unnecessary checks.

**Log Message**
In regards to the log message, do you know if a log message is necessary or would be helpful?
Basically, can we think of any use-cases where the log message would be helpful?

From my understanding, the main goal of the logic is for a slave node to retrieve the scheduled jobs from the master node.

To me, a deciding factor would be the following question: Is it okay for the retrieved schedulerResponse to not have a trigger?

If that's okay and is an expected scenario, then I don't think a log message would be necessary.

Again, I haven't really investigated this issue, so I'm not exactly sure if that's the actual scenario.
Can you look into whether this is the case?

Hopefully that helps, feel free to let me know if you have any further questions.
Thanks!

**Comment by ericyanLr** [comment]
Closing this pr for now, feel free to re-send.

**Comment by istvansajtos** [comment]
Hi Eric,

Thank you for sharing your train of thoughts. I will resend the PR once I clarified the questions that came up.

Regards!
